### PR TITLE
[kube-prometheus-stack] Add null receiver to InfoInhibitor rule in default AlertManager configuration

### PR DESCRIPTION
**Pull request by jnoordsij**
#### What this PR does / why we need it:
Add `null` as receiver for the InfoInhibitor rule, and also replace the deprecated `match` syntax with a `matchers` directive. Based on the discussion in #1773.

This prevents the InfoInhibitor rule from causing noise on other receivers, as recommended in the rule description.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zanhsieh — DISMISSED** [review]

**Review by zanhsieh — DISMISSED** [review]

**Comment by jemag** [comment]
would it make more sense to define the inhibit rule as well? Since I believe this is the intent of this alert.

Also in general wondering if inhibit_rules should be defined by default, since upstream kube-prometheus project is doing it : https://github.com/prometheus-operator/kube-prometheus/blob/0ac0d7f9bf5ab1de3e596ff3bdd4b36c52e27267/jsonnet/kube-prometheus/components/alertmanager.libsonnet#L30-L42

**Review by zanhsieh — APPROVED** [review]

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 35.2.0
+version: 35.2.1
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]:
```suggestion
version: 35.2.1
```

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Comment by gkarthiks** [comment]
@jnoordsij  @zanhsieh Thank you for the contribution. LGTM. 

A small request, next time please squash your commits into one. So that it would be easy for review.

**Review by gkarthiks — APPROVED** [review]

**Comment by jemag** [comment]
@gkarthiks any particular reason to merge this without the inhibit rule? 

My understanding is that the sole purpose of this alert is to inhibit other `Info` alerts, as is mentioned in the alert description:
```
...
          This alert should be routed to a null receiver and configured to inhibit alerts with severity="info".
``` 

As far as I am aware, now it is only routed to null but not actually inhibiting anything. It will also not notify users of the chart to do the proper modifications to their own inhibit rules since, by default, it will be routed to the null receiver, effectively rendering it useless to unaware users.

**Comment by jnoordsij** [comment]
> @gkarthiks any particular reason to merge this without the inhibit rule?
> 
> My understanding is that the sole purpose of this alert is to inhibit other `Info` alerts, as is mentioned in the alert description:
> 
> ```
> ...
>           This alert should be routed to a null receiver and configured to inhibit alerts with severity="info".
> ```
> 
> As far as I am aware, now it is only routed to null but not actually inhibiting anything. It will also not notify users of the chart to do the proper modifications to their own inhibit rules since, by default, it will be routed to the null receiver, effectively rendering it useless to unaware users.

I think this is exactly what the current rule definition (https://github.com/prometheus-community/helm-charts/blob/e2af4613e9c79f3f663c697c2e33b711a676b142/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/general.rules.yaml#L82) already does, though I'm not sure on the particulars. The `InfoInhibitor` alert has been firing quite a few times in my setup, without additional config. So it seems to be already in place...

**Comment by jnoordsij** [comment]
@jemag ahhh I see the difference now, I was a little confused about the difference between the alert firing and it actually inhibiting and thus muting the `info` alerts. I created #2114 to implement the actually inhibit rules in the same way kube prometheus does it, would love your feedback on that!

**Pull request merged**